### PR TITLE
fix: improve lightning payment balance assertions in cli-test.sh

### PR DESCRIPTION
**Pull request by justinmoon**
I commented-out some balance assertions in https://github.com/fedimint/fedimint/pull/2052 because they didn't make sense to me and I wanted to revisit them. But I forgot to. Luckily @m1sterc001guy [caught it in review](https://github.com/fedimint/fedimint/pull/2052#discussion_r1151168497). I merged that PR and fix the issue here.

TLDR: cli-test.sh sends 100,000 msats from user to external lightning node via gateway. Looking at the old code, I mistakenly assumed that the starting balance of the gateway was 100,000,000 msats so checking that the final balance of the gateway increased to 100,100,000 seemed wrong -- what about the [hard-coded 1% fee](https://github.com/fedimint/fedimint/blob/428166662c4ccf7dbb415e6b93ee16fb2d52ade3/client/client-lib/src/ln/mod.rs#L84-L100)?

I have been confused by this before and didn't investigate. What I discovered was that the gateway's initial balance was actually 99,999,000 msats and not 100,000,000 msats. So as expected the 1% fee was being taken.

Refactored the tests to make assertions by subtracting initial and final balances so this is clearer.

### CODE REVIEW

**Review by dpc — APPROVED** [review]

**Review by okjodom — APPROVED** [review]
✅

**Review by m1sterc001guy — APPROVED** [review]

**Review by dpc — CHANGES_REQUESTED** [review]
Please add `>&2 echo "Assertion failed: blablabla"`

File: scripts/cli-test.sh [hunk 1]
```diff
@@ -60,17 +60,22 @@ await_lightning_node_block_processing
...
-# LN_GATEWAY_BALANCE="$($FM_GATEWAY_CLI balance $FED_ID | jq -e -r '.balance_msat')"
-# [[ "$LN_GATEWAY_BALANCE" = "100100000" ]]
+
 INVOICE_STATUS="$($FM_LNCLI lookupinvoice $PAYMENT_HASH | jq -e -r '.state')"
 [[ "$INVOICE_STATUS" = "SETTLED" ]]
 
+# Assert balances changed by 100000 msat (amount sent) + 1000 msat (fee)
+FINAL_CLIENT_BALANCE="$($FM_MINT_CLIENT info | jq -e -r '.total_amount')"
+FINAL_GATEWAY_BALANCE="$($FM_GWCLI_CLN balance $FED_ID | jq -e -r '.balance_msat')"
+[[ "$(($INITIAL_CLIENT_BALANCE - $FINAL_CLIENT_BALANCE))" = "101000" ]]
```
[the remark this answers is not in the document]
Comment (justinmoon) [reply scripts/cli-test.sh]: Fixed

**Review by dpc — DISMISSED** [review]

File: scripts/cli-test.sh [hunk 2]
```diff
@@ -51,7 +51,7 @@ until [ "$($FM_BTC_CLIENT getreceivedbyaddress $PEG_OUT_ADDR 0)" == "0.00000500"
 done
 mine_blocks 10
 RECEIVED=$($FM_BTC_CLIENT getreceivedbyaddress $PEG_OUT_ADDR)
-[[ "$RECEIVED" = "0.00000500" ]]
+[[ "$RECEIVED" != "0.00000500" ]];
```
Comment (jkitman) [inline scripts/cli-test.sh:54]: Why does this assert `!=` instead of the exact `=` amount?

Comment (dpc) [reply scripts/cli-test.sh:54]: Yeah, what happened here. :D

Also will fail silently which is not good.

Comment (dpc) [reply scripts/cli-test.sh:54]: We now know that we had a fiat CI again.

**Comment by justinmoon** [comment]
Pushed a whitespace commit to see if the erroneous `!=` will trigger CI failure now.

File: scripts/cli-test.sh [hunk 3]
```diff
@@ -51,7 +51,7 @@ until [ "$($FM_BTC_CLIENT getreceivedbyaddress $PEG_OUT_ADDR 0)" == "0.00000500"
 done
 mine_blocks 10
 RECEIVED=$($FM_BTC_CLIENT getreceivedbyaddress $PEG_OUT_ADDR)
-[[ "$RECEIVED" = "0.00000500" ]]
+[[ "$RECEIVED" == "0.00000500" ]];
```
Comment (dpc) [inline scripts/cli-test.sh:54]: Well, if we're already touching it ... why not a proper error message instead of failing silently? :)

**Comment by justinmoon** [comment]
Passing CI now

**Review by dpc — DISMISSED** [review]

File: scripts/cli-test.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 # Runs a CLI-based integration test
 
-set -euo pipefail
+set -euxo pipefail
```
Comment (elsirion) [inline scripts/cli-test.sh:4]: Sneaking in `-x` again? :P

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
Requires rebase :(

**Comment by justinmoon** [comment]
Closing this because it's fixed in https://github.com/fedimint/fedimint/pull/2147. CI got the better of me this time 😂

**Pull request closed without merging**